x-pack/plugins/infra/public/legacy_singletons.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { CoreSetup, CoreStart } from 'kibana/public';
+
+let npSetup: CoreSetup;
+let npStart: CoreStart;
+
+export function registerSetupSingleton(setup: CoreSetup) {
+  npSetup = setup;
+}
```

Comment:
nit: setup seems to be unused and can be removed. Setup is also unlikely to be shared deeply in component trees the way start would be.

Reply:
Yeah, totally unused, I'll remove this 👍